api/src/client_structs.rs

```diff
@@ -0,0 +1,212 @@
+use serde::{Deserialize, Serialize};
+#[allow(dead_code)]
+#[allow(non_snake_case)] // backwards compatibility
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CutCards {
+    pub Player: ClientCard,
+    pub Computer: ClientCard,
+}
+impl CutCards {
+    pub fn new(p_index: usize, c_index: usize) -> CutCards {
+        CutCards {
+            Player: ClientCard::from_card(Card::from_index(p_index), "Player".to_string()),
+            Computer: ClientCard::from_card(Card::from_index(c_index), "Computer".to_string()),
+        }
+    }
+}
+#[allow(dead_code)]
+#[allow(non_snake_case)] // backwards compatibility
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CutCardResponse {
+    pub CutCards: CutCards,
+    pub RepeatUrl: String,
+}
+
+#[allow(non_snake_case)] // backwards compatibility
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ScoreInfo {
+    pub ScoreName: CombinationName,
+    pub Score: u32,
+    pub Cards: Vec<ClientCard>,
+}
+
+impl ScoreInfo {
+    pub fn from_combination(combi: Combination) -> ScoreInfo {
+        let mut client_cards: Vec<ClientCard> = Vec::<ClientCard>::default();
+        for card in combi.cards.into_iter() {
+            client_cards.push(ClientCard::from_card(card, "unknown".to_string()));
+        }
+
+        ScoreInfo {
+            ScoreName: combi.name,
+            Score: combi.points,
+            Cards: client_cards,
+        }
+    }
+}
+
+#[allow(non_snake_case)] // backwards compatibility
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ScoreResponse {
+    pub Score: u32,
+    pub ScoreInfo: Vec<ScoreInfo>,
+}
+
+impl ScoreResponse {
+    pub fn default() -> ScoreResponse {
+        ScoreResponse {
+            Score: 0,
+            ScoreInfo: Vec::<ScoreInfo>::default(),
+        }
+    }
+
+    pub fn from_score(score: Score) -> ScoreResponse {
+        let mut scores = Vec::<ScoreInfo>::new();
+        score.combinations.into_iter().for_each(|combi| {
+            scores.push(ScoreInfo::from_combination(combi));
+        });
+        ScoreResponse {
+            Score: score.total_score,
+            ScoreInfo: scores,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ParsedHand {
+    pub hand: Vec<Card>,
+}
+
+impl ParsedHand {
+    pub fn from_string(csv: String) -> Result<ParsedHand, CribbageError> {
+        let mut hand: Vec<Card> = Vec::<Card>::new();
+        let tokens: Vec<&str> = csv.split(",").collect();
+        for token in tokens {
+            let _ = match Card::from_string(token) {
+                Ok(card) => {
+                    hand.push(card);
+                }
+                Err(e) => {
+                    return Err(e);
+                }
+            };
+        }
+
+        Ok(ParsedHand { hand })
+    }
+}
+
+//
+//  converts a CSV of cards into a Vec<Card>
+//  returns: Vec<Card> or HttpResponse::BadRequest
+#[allow(unused_macros)]
+#[macro_export]
+macro_rules! csv_to_cards {
+    ($csv: expr) => {{
```

Comment:
this one did not.  but get_repeat_url does (won't build w/o them).  if you need them and dn't have them you get this error:

`let` expressions in this position are experimental
see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
you can write `matches!(<expr>, <pattern>)` instead of `let <pattern> = <expr>`rustcE0658